### ISSUE TITLE
DNSSEC allow-downgrade not working (in Deutsche Bahn WIFI)

### ISSUE THREAD

**Issue by paulmenzel** [systemd/systemd#15073]
With Debian Sid/unstable and

    $ systemd 244 (244.3-1)
    +PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid

running the default *systemd-resolved* configuration, `DNSSEC=allow-downgrade` does not work in the WIFI of the Deutsche Bahn *DB Bahn WIFI*. 

```
Global
       LLMNR setting: yes
MulticastDNS setting: yes
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes
  Current DNS Server: 172.18.0.1
         DNS Servers: 172.18.0.1
          DNSSEC NTA: 10.in-addr.arpa
                      16.172.in-addr.arpa
                      168.192.in-addr.arpa
                      17.172.in-addr.arpa
                      18.172.in-addr.arpa
                      19.172.in-addr.arpa
                      20.172.in-addr.arpa
                      21.172.in-addr.arpa
                      22.172.in-addr.arpa
                      23.172.in-addr.arpa
                      24.172.in-addr.arpa
                      25.172.in-addr.arpa
                      26.172.in-addr.arpa
                      27.172.in-addr.arpa
                      28.172.in-addr.arpa
                      29.172.in-addr.arpa
                      30.172.in-addr.arpa
                      31.172.in-addr.arpa
                      corp
                      d.f.ip6.arpa
                      home
                      internal
                      intranet
                      lan
                      local
                      private
                      test

Link 3 (wlan0)
      Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6
DefaultRoute setting: yes
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes
  Current DNS Server: 172.18.0.1
         DNS Servers: 172.18.0.1
          DNS Domain: ~.

Link 2 (eno1)
      Current Scopes: none
DefaultRoute setting: no
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: allow-downgrade
    DNSSEC supported: yes
```

DNS queries hang with `systemd-resolve` (also in the browser *Firefox*), but work with `host`.

    $ systemd-resolve www.molgen.mpg.de
    Ctrl + c
    $ host www.molgen.mpg.de
    www.molgen.mpg.de is an alias for npsw-www.mpg.de.
    npsw-www.mpg.de has address 134.76.31.205

Stopping `systemd-resolved` or using `DNSSEC=no` in the systemd-resolved configuration fixes the issue.

```
Global
       LLMNR setting: yes
MulticastDNS setting: yes
  DNSOverTLS setting: no
      DNSSEC setting: no
    DNSSEC supported: no
  Current DNS Server: 172.18.0.1
         DNS Servers: 172.18.0.1
          DNSSEC NTA: 10.in-addr.arpa
                      16.172.in-addr.arpa
                      168.192.in-addr.arpa
                      17.172.in-addr.arpa
                      18.172.in-addr.arpa
                      19.172.in-addr.arpa
                      20.172.in-addr.arpa
                      21.172.in-addr.arpa
                      22.172.in-addr.arpa
                      23.172.in-addr.arpa
                      24.172.in-addr.arpa
                      25.172.in-addr.arpa
                      26.172.in-addr.arpa
                      27.172.in-addr.arpa
                      28.172.in-addr.arpa
                      29.172.in-addr.arpa
                      30.172.in-addr.arpa
                      31.172.in-addr.arpa
                      corp
                      d.f.ip6.arpa
                      home
                      internal
                      intranet
                      lan
                      local
                      private
                      test

Link 3 (wlan0)
      Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6
DefaultRoute setting: yes
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: no
    DNSSEC supported: no
         DNS Servers: 172.18.0.1
          DNS Domain: ~.

Link 2 (eno1)
      Current Scopes: none
DefaultRoute setting: no
       LLMNR setting: yes
MulticastDNS setting: no
  DNSOverTLS setting: no
      DNSSEC setting: no
    DNSSEC supported: no
```

Any ideas. Also, why the Ethernet device, where *nothing* is connected to, says DNSSEC is supported?

**Comment by shibumi** [comment]
Can you show some logs of systemd-resolved?

`sudo journalctl -u systemd-resolved`.

**Comment by chiefnoah** [comment]
I also encountered this on Arch Linux with systemd 245. Previously working configuration stopped working after updating. `journalctl` reports a bunch of errors resolving domains:

```
ar 13 13:03:03 komoe systemd[1]: Starting Network Name Resolution...
Mar 13 13:03:03 komoe systemd-resolved[76601]: Positive Trust Anchors:
Mar 13 13:03:03 komoe systemd-resolved[76601]: . IN DS 20326 8 2 e06d44b80b8f1d39a95c0b0d7c65d08458e880409bbc683457104237c7f8ec8d
Mar 13 13:03:03 komoe systemd-resolved[76601]: Negative trust anchors: 10.in-addr.arpa 16.172.in-addr.arpa 17.172.in-addr.arpa 18.172.in-addr.arpa 19.172.in-addr.arpa 20.>
Mar 13 13:03:03 komoe systemd-resolved[76601]: Using system hostname 'komoe'.
Mar 13 13:03:03 komoe systemd[1]: Started Network Name Resolution.
Mar 13 13:03:06 komoe systemd-resolved[76601]: DNSSEC validation failed for question packetlostandfound.us IN DS: no-signature
Mar 13 13:03:06 komoe systemd-resolved[76601]: DNSSEC validation failed for question packetlostandfound.us IN SOA: no-signature
```
I can provide my configuration, but it's since changed to set `DNSSEC=` to `no`, which resolved the issue.

**Comment by paulmenzel** [comment]
@chiefnoah, in my case it has never worked with the recent releases. Just to clarify, it’s a regression for you from systemd 244?

“it's since changed to set DNSSEC= to no, which resolved the issue.” in your last sentence is quite confusing. Could you rephrase?

**Comment by paulmenzel** [comment]
@shibumi, sure.

```
Mär 10 10:19:05.695046 ersatz systemd[1]: Starting Network Name Resolution...
Mär 10 10:19:06.145403 ersatz systemd-resolved[444]: Positive Trust Anchors:
Mär 10 10:19:06.145421 ersatz systemd-resolved[444]: . IN DS 20326 8 2 e06d44b80b8f1d39a95c0b0d7c65d08458e880409bbc683457104237c7f8ec8d
Mär 10 10:19:06.145481 ersatz systemd-resolved[444]: Negative trust anchors: 10.in-addr.arpa 16.172.in-addr.arpa 17.172.in-addr.arpa 18.172.in-addr.ar
pa 19.172.in-addr.arpa 20.172.in-addr.arpa 21.172.in-addr.arpa 22.172.in-addr.arpa 23.172.in-addr.arpa 24.172.in-addr.arpa 25.172.in-addr.arpa 26.172.
in-addr.arpa 27.172.in-addr.arpa 28.172.in-addr.arpa 29.172.in-addr.arpa 30.172.in-addr.arpa 31.172.in-addr.arpa 168.192.in-addr.arpa d.f.ip6.arpa cor
p home internal intranet lan local private test
Mär 10 10:19:06.185233 ersatz systemd-resolved[444]: Using system hostname 'ersatz'.
Mär 10 10:19:06.349073 ersatz systemd[1]: Started Network Name Resolution.
Mär 10 10:19:26.359288 ersatz systemd-resolved[444]: Using degraded feature set (UDP+EDNS0+DO) for DNS server 172.18.0.1.
Mär 10 10:23:02.374503 ersatz systemd-resolved[444]: DNSSEC validation failed for question detectportal.firefox.com IN A: failed-auxiliary
Mär 10 10:23:02.374656 ersatz systemd-resolved[444]: DNSSEC validation failed for question push.services.mozilla.com IN A: failed-auxiliary
Mär 10 10:23:02.375240 ersatz systemd-resolved[444]: DNSSEC validation failed for question safebrowsing.googleapis.com IN A: failed-auxiliary
Mär 10 10:23:02.375501 ersatz systemd-resolved[444]: DNSSEC validation failed for question firefox.settings.services.mozilla.com IN A: failed-auxiliar
y
Mär 10 10:23:02.375931 ersatz systemd-resolved[444]: DNSSEC validation failed for question www.wifionice.de IN A: failed-auxiliary
Mär 10 10:23:07.609775 ersatz systemd-resolved[444]: Using degraded feature set (UDP+EDNS0+DO) for DNS server 172.18.0.1.
Mär 10 10:30:30.859494 ersatz systemd-resolved[444]: DNSSEC validation failed for question detectportal.firefox.com IN A: failed-auxiliary
Mär 10 10:35:13.359467 ersatz systemd-resolved[444]: DNSSEC validation failed for question detectportal.firefox.com IN A: failed-auxiliary
Mär 10 10:37:25.859385 ersatz systemd-resolved[444]: DNSSEC validation failed for question blog.fefe.de IN A: failed-auxiliary
Mär 10 10:38:24.248717 ersatz systemd-resolved[444]: Failed to emit notification about changed property CurrentDNSServer: Transport endpoint is not co
nnected
Mär 10 10:38:24.248881 ersatz systemd-resolved[444]: DNSSEC validation failed for question detectportal.firefox.com IN A: failed-auxiliary
Mär 10 10:38:24.248914 ersatz systemd-resolved[444]: Failed to send hostname reply: Transport endpoint is not connected
```

**Comment by chiefnoah** [comment]
Correct, my configuration worked with systemd 244 and stopped working after updating to 245.

> “it's since changed to set DNSSEC= to no, which resolved the issue.” in your last sentence is quite confusing. Could you rephrase?

I meant I can provide my configuration, but I've changed it to disable DNSSEC so it's not in the state that it was in when broken and may not be helpful.

**Comment by shibumi** [comment]
I am pretty sure this problem has to do with a not valid DNSSEC chain.
Can you guys use https://dnsviz.net/ next time when you are in the ICE?
(and make a screenshot of it). Just use a domain that fails for you in the ICE.
I expect broken infrastructure of Deutsche Bahn ;)

**Comment by paulmenzel** [comment]
How is the result in https://dnsviz.net/ dependent on the network you are in?

**Comment by darkdragon-001** [comment]
Duplicate of #11240 ?

**Comment by pemensik** [comment]
Please when would you travel next time, try please delv tool from bind-utils. It tries validation on client side, outside systemd. And it reports coherent messages, where is validation broken.

```
delv +cd +vtrace detectportal.firefox.com
# Then compare it with DHCP resolver directly
delv +cd +vtrace @dns-ip-provided-by-dhcp detectportal.firefox.com
```

It needs dnssec enabled. Unfortunately, there is no simple way to just disable validation, which should help here.

If only few names stopped working, I would recommend setting ``resolvectl nta <iface> <domain>``.

**Comment by poettering** [comment]
Duplicate of #11240. Let's continue things there.
